Shape_detection/include/CGAL/Shape_detection/Region_growing/Region_growing_on_polygon_mesh/Polyline_graph.h

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2020 GeometryFactory SARL (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Dmitry Anisimov
+//
+
+#ifndef CGAL_SHAPE_DETECTION_REGION_GROWING_POLYGON_MESH_POLYLINE_GRAPH_H
+#define CGAL_SHAPE_DETECTION_REGION_GROWING_POLYGON_MESH_POLYLINE_GRAPH_H
+
+#include <CGAL/license/Shape_detection.h>
+
+// Internal includes.
+#include <CGAL/Shape_detection/Region_growing/internal/property_map.h>
+
+namespace CGAL {
+namespace Shape_detection {
+namespace Polygon_mesh {
+
+  /*!
+    \ingroup PkgShapeDetectionRGOnMesh
+
+    \brief Polygon mesh edges connected into a graph.
+
+    This class returns all edges, which form polylines splitting the polygon mesh
+    being a `PolygonMesh` into planar regions.
+
+    \tparam GeomTraits
+    a model of `Kernel`
+
+    \tparam PolygonMesh
+    a model of `FaceListGraph`
+
+    \tparam FaceRange
+    a model of `ConstRange` whose iterator type is `RandomAccessIterator` and
+    value type is the face type of a polygon mesh
+
+    \tparam EdgeRange
+    a model of `ConstRange` whose iterator type is `RandomAccessIterator` and
+    value type is the edge type of a polygon mesh
+
+    \tparam VertexToPointMap
+    a model of `LValuePropertyMap` whose key type is the vertex type of a polygon mesh and
+    value type is `Kernel::Point_3`
+
+    \cgalModels `NeighborQuery`
+  */
+  template<
+  typename GeomTraits,
+  typename PolygonMesh,
+  typename FaceRange = typename PolygonMesh::Face_range,
+  typename EdgeRange = typename PolygonMesh::Edge_range,
+  typename VertexToPointMap = typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::type>
+  class Polyline_graph {
```

Comment:
I'm not sure why you need this class for. Only to creates the segment network or also to store the result of the line fitting?

Reply:
I will change the API of this class. But the class has a twofold task: create a network of segments and it is also a model of the concept `Neighbor_query()` that is required for running region growing on the above network.